concurrency/conc2_packed_task.cpp

```diff
@@ -70,10 +69,24 @@ std::vector<char> readFileContent(std::filesystem::path file_name) {
         throw std::runtime_error(std::string("vector push_back method failed") + e.what());
     }
 
-    if (bytes_read == -1) {
-        throw std::system_error(std::make_error_code(static_cast<std::errc>(errno)), "Read syscall failed");
+    int times_retried = 0;
+    try {
+        while (bytes_read == -1 && times_retried < 3) {
+            if (bytes_read == -1) {
+                if (errno & EAGAIN || errno & EAGAIN || errno & EAGAIN) {
+                    ++times_retried;
+                } else {
+                    throw std::system_error(std::make_error_code(static_cast<std::errc>(errno)), "Read syscall failed");
+                }
+            }
+
+            while ( (bytes_read = read(fd, buffer.data(), BUFFER_SIZE)) > 0) {
+                data_read.insert(data_read.end(), buffer.begin(), buffer.begin() + bytes_read);
+            } 
+        }
```

Comment:
This retry mechanism can be written in a better way.

Reply:
done